shardy/dialect/sdy/ir/attrs.td

```diff
@@ -746,6 +790,17 @@ def Sdy_OpShardingRule : AttrDef<Sdy_Dialect, "OpShardingRule"> {
     these ops, so a user must tell it how. When it is a custom rule, then the
     rule is always preserved/never removed. `is_custom_rule` can only be true
     for `stablehlo.custom_call` ops.
+
+    **Constraints:**
+    - Elements in `reduction_factors` and `need_replication_factors` must
+      satisfy the constraints of `DimMappingAttr`.
+    - Number of operand/result mappings must match the number of
+      operands/results of the op.
+    - There is at least one mapping (can't have a rule for an op with no
+      operands/results).
+    - Rank of each `TensorMappingAttr` matches the rank of the corresponding
+      tensor type.
+    - No duplicate factor indices in each `TensorMappingAttr`.
```

Comment:
This should go into DimMappingAttr (no duplicate factors in each `DimMappingAttr`)